message-service/src/main/java/com/community/tools/discord/CommandHandler.java

```diff
@@ -0,0 +1,105 @@
+package com.community.tools.discord;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Getter
+@Component
+public class CommandHandler {
+
+  private final Map<String, Object> namesCommandsMap;
+
+  /**
+   * Constructor, which gathers all commands from app context.
+   * @param context Spring's application context
+   */
+  public CommandHandler(ApplicationContext context) {
+    this.namesCommandsMap = context.getBeansWithAnnotation(Command.class).entrySet().stream()
+        .filter(CommandHandler::hasRunnableMethod)
```

Comment:
would be better to have a base class for a disc commands to autowire them witout this constructor

Reply:
I aggree, reflection is unnecessary here. This tool should be your last resort in almost all cases